test/gtest/uct/ib/test_cq_moderation.cc

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2017.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+extern "C" {
+#include <uct/api/uct.h>
+}
+#include <uct/uct_test.h>
+#include <ucs/time/time.h>
+#include <poll.h>
+#include <infiniband/verbs.h>
+
+#define TEST_PERIOD (3ul * UCS_USEC_PER_SEC) /* wait for 3 secs to get statistics */
+#define NSEC_PER_USEC (UCS_NSEC_PER_SEC / UCS_USEC_PER_SEC)
+
+#define MODERATION_PERIOD 1000 /* usecs */
+#define EVENT_LIMIT (10 * TEST_PERIOD / MODERATION_PERIOD / NSEC_PER_USEC)
+#define MAX_REPEATS 1000
+
+#define _STR(_arg) #_arg
+#define TO_STR(_val) _STR(_val)
+
+class test_uct_cq_moderation : public uct_test {
+protected:
+
+    void init() {
+        uct_test::init();
+
+        if (!(GetParam()->tl_name == "rc" || GetParam()->tl_name == "rc_mlx5" ||
+              GetParam()->tl_name == "ud" || GetParam()->tl_name == "ud_mlx5" ||
+              GetParam()->tl_name == "dc" || GetParam()->tl_name == "dc_mlx5")) {
+            /* dc_mlx5 is masked due to unknown issue - CQ moderation doesn't work on DC */
+            UCS_TEST_SKIP_R("unsupported");
+        }
+
+        if (RUNNING_ON_VALGRIND) {
+            UCS_TEST_SKIP_R("skipping on valgrind");
+        }
+
+        set_config("IB_TX_CQ_MODERATION=1");
+        if (!(GetParam()->tl_name == "ud" || GetParam()->tl_name == "ud_mlx5")) {
```

Comment:
better have a positive check (e.g if the tl is rc/dc/rc_mlx5/dc_mlx5)

Reply:
done